src/azureFunction/azureFunctionUtils.ts

```diff
@@ -0,0 +1,176 @@
+}
+
+/**
+ * copied and modified from vscode-azurefunctions extension
+ * @param localSettingsPath full path to local.settings.json
+ * @returns settings in local.settings.json. If no settings are found, returns default "empty" settings
+ */
+export async function getLocalSettingsJson(localSettingsPath: string): Promise<ILocalSettingsJson> {
+	if (await fse.pathExists(localSettingsPath)) {
+		const data: string = (await fse.readFile(localSettingsPath)).toString();
+		if (/[^\s]/.test(data)) {
+			try {
+				return parseJson(data);
+			} catch (error) {
+				throw new Error(constants.failedToParse(error.message));
+			}
+		}
+	}
+
+	return {
+		IsEncrypted: false // Include this by default otherwise the func cli assumes settings are encrypted and fails to run
+	};
+}
+
+/**
+ * Adds a new setting to a project's local.settings.json file
+ * modified from setLocalAppSetting code from vscode-azurefunctions extension
+ * @param projectFolder full path to project folder
+ * @param key Key of the new setting
+ * @param value Value of the new setting
+ * @returns true if successful adding the new setting, false if unsuccessful
+ */
+export async function setLocalAppSetting(projectFolder: string, key: string, value: string): Promise<boolean> {
+	const localSettingsPath: string = path.join(projectFolder, constants.azureFunctionLocalSettingsFileName);
+	const settings: ILocalSettingsJson = await getLocalSettingsJson(localSettingsPath);
+
+	settings.Values = settings.Values || {};
+	if (settings.Values[key] === value) {
+		// don't do anything if it's the same as the existing value
+		return true;
+	} else if (settings.Values[key]) {
+		const result = await vscode.window.showWarningMessage(constants.settingAlreadyExists(key), { modal: true }, constants.yesString);
+		if (result !== constants.yesString) {
+			// key already exists and user doesn't want to overwrite it
+			return false;
+		}
+	}
+
+	settings.Values[key] = value;
+	await fse.writeJson(localSettingsPath, settings, { spaces: 2 });
+
+	return true;
+}
+
+/**
+ * Adds specified package to Azure Functions project the specified file is a part of
+ * @param filePath uri of file to find the containing AF project of to add package reference to
+ * @param packageName package to add reference to
+ * @param packageVersion optional version of package. If none, latest will be pulled in
+ */
+export async function addPackageToAFProjectContainingFile(fileUri: vscode.Uri, packageName: string, packageVersion?: string): Promise<void> {
+	try {
+		const project = await getAFProjectContainingFile(fileUri);
+
+		// if no AF projects were found, an error gets thrown from getAFProjectContainingFile(). This check is temporary until
+		// multiple AF projects in the workspace is handled. That scenario returns undefined and shows an info message telling the
+		// user to make sure their project has the package reference
+		if (project) {
+			await this.addPackage(project, packageName, packageVersion);
```

Comment:
Is this correct?

1. I don't see the `addPackage` defined
2. You're using `this` but this is at the module level so there's no "this" to bind to

Reply:
removed this found a simpler solution for the current functionality.